### DISCUSSION TITLE
Exit the optimization loop based on a different criteria

### DISCUSSION THREAD

**Discussion by uttamcadambi07** [mdolab/dafoam#435] [General]
Hi.

I use IPOPT. I would like to calculate a term myself and use that to exit out of the optimization loop. I do not want to touch the IPOPT code because i feel I may mess it up.

Is there a way to give some sort of condition on DAFoam that will allow me to terminate the optimization process? This is what the code looks like in runScript.py
```
if args.task == "opt":

    optProb = Optimization("opt", objFun=optFuncs.calcObjFuncValues, comm=gcomm)
    DVGeo.addVariablesPyOpt(optProb)
    DVCon.addConstraintsPyOpt(optProb)

    optProb.addObj("FI", scale=1)

    if gcomm.rank == 0:
        print(optProb)

    DASolver.runColoring()

    opt = OPT(args.opt, options=optOptions)
    histFile = "./%s_hist.hst" % args.opt
    sol = opt(optProb, sens=optFuncs.calcObjFuncSens, storeHistory=histFile)
    if gcomm.rank == 0:
        print(sol)
```

So ideally I am looking at the line ```sol = opt(optProb, sens=optFuncs.calcObjFuncSens, storeHistory=histFile)```. Any suggestions

EDIT: I must mention that the term I would like to use will be calculated in DASolver class. Think of it as a L1 norm of velocity field.

**Comment by friedenhe** [comment]
There is no easy fix for this. A workaround is to use the `FatalErrorIn` function to exit DAFoam when the self-defined criterion is met.
